### PR TITLE
Refactor MultiBlock and slice_along_axis

### PR DESCRIPTION
**Pull request by adeak**
Some refactors that came up during https://github.com/pyvista/pyvista/pull/1482. Closes https://github.com/pyvista/pyvista/issues/1494.

- [x] https://github.com/pyvista/pyvista/pull/1482#discussion_r664015140 (refactor `MultiBlock.bounds`)
- [x] https://github.com/pyvista/pyvista/pull/1482#discussion_r664019043 (refactor `MultiBlock.volume`)
- [x] ~~https://github.com/pyvista/pyvista/pull/1482#discussion_r664022363 (refactor `MultiBlock.get_block_name`)~~ we can't actually refactor this because we'd need `.keys()` and `.keys()` is defined in terms of `get_block_name`...
- [x] https://github.com/pyvista/pyvista/pull/1482#discussion_r664094024 (refactor `DataSetFilters.slice_along_axis`)

### CODE REVIEW

File: pyvista/core/composite.py [hunk 1]
```diff
@@ -135,25 +135,12 @@ def bounds(self) -> List[float]:
...
-            for a in range(3):
-                bounds = update_bounds(a, bnds, bounds)
-
-        return bounds
+        # apply reduction of min and max over each block
+        all_bounds = [block.bounds for block in self if block]
+        minima = np.minimum.reduce(all_bounds)[::2]
+        maxima = np.maximum.reduce(all_bounds)[1::2]
+        # interleave minima and maxima for bounds
+        return np.stack([minima, maxima]).ravel('F').tolist()
```
Comment (akaszynski) [inline pyvista/core/composite.py:143]: Woa, that's way cleaner.

File: pyvista/core/filters/data_set.py [hunk 2]
```diff
@@ -511,17 +511,30 @@ def slice_along_axis(dataset, n=5, axis='x', tolerance=None,
             axis index (``0``, ``1``, or ``2``).
 
         tolerance : float, optional
-            The tolerance to the edge of the dataset bounds to create the
-            slices.
+            The tolerance to the edge of the dataset bounds to create
+            the slices. The ``n`` slices are placed equidistantly with
+            an absolute padding of ``tolerance`` inside each side of the
+            ``bounds``- along the specified axis. Defaults to 1% of the
```
Comment (akaszynski) [inline pyvista/core/filters/data_set.py:517]: Is the `-` meant to be here?

Comment (adeak) [reply pyvista/core/filters/data_set.py:517]: Absolutely not! Thanks for spotting it :)

Comment (adeak) [reply pyvista/core/filters/data_set.py]: Fixed in https://github.com/pyvista/pyvista/pull/1506/commits/31dd0f2ef1ec1c32b9235880938fdb092915c05f

**Review by akaszynski — APPROVED** [review]
LGTM.  Thanks for the refactor!

**Pull request merged**